### ISSUE TITLE
Margin below the description

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [Ks89/angular-modal-gallery#133]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## Current behavior
The description element on top of the image in the modal shows a few pixels off from the bottom of the image. The number of pixels depends on the typographic settings of the page. This behaviour is caused by:
1. The img element inside the figure element hasn't been set to display:block, which means its an inline element and therefor a line-height. The line height is causing additional margin below the image (which becomes apparent if you add a border around the figure element). The size of the margin depends on the (inherited) line height of the element. 
2. The figcaption is styled with a bottom of 5px, which is probably done to overcome the margin between the figure and image (but doesn't look great if you the margin is smaller). 

## Expected behavior
The description element should be positioned at the exact bottom of the image (I assume this is the intend).

## Minimal reproduction of the problem with instructions
All demo's come with this behaviour. Let me know if you need more info. 

## What is the motivation / use case for changing the behavior?
To make it look nice :)

Browser:
- [x ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [x] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

**Comment by tobi-or-not-tobi** [comment]
I've been able to fix this in project by adding the following to my style.scss:

```

ks-current-image {
    figure {
        img {
            display: block;
        }
        figcaption {
            bottom: 0px !important;
        }
    }
}
```

**Comment by Ks89** [comment]
Mmm, interesting. I'll check this issue ASAP.

**Comment by Ks89** [comment]
I fixed this issue, however I cannot release it because for some users this could be a breaking change.
So, I'll add other options to description object to be able to change color and set margins.
By default, I'll leave 5px, so I can release it as minor release 5.x.x, instead of 6.x.x.

Give some days to add the other input properties.

**Comment by Ks89** [comment]
I updated Description interface with these values (with defaults):

```
margin: {
        marginTop: '0px',
        marginBottom: '5px',
        marginLeft: '0px',
        marginRight: '0px'
      },
      style: {
        bgColor: 'rgba(0, 0, 0, .5)',
        textColor: 'white'
      }
```

which other params do you need?

**Comment by tobi-or-not-tobi** [comment]
To begin with, as soon as you add the display:block to the img element, you make a change to the positioning of the description element. I think its ok, just want to make sure you're aware. 

Regarding the properties, it would be nice to pass a style class, so that the style rules can be freely defined inside css. I'm not sure however if we can bypass the encapsulation with this setup. 

If you use the style properties, I'd add margin properties _inside_ style; to me the margins are also style properties.

**Comment by Ks89** [comment]
I released this feature as 5.4.0-beta.1

If you want to try it, use `npm i --save angular-modal-gallery@beta`

**Comment by Ks89** [comment]
' believe no one likes to default to 5px from bottom.' this could be wrong because you are the only one that asked me to fix this. Probably, many people could consider this as a feature (thy are crazy, ok, but it's possibile). 
Since I'm compliant with semver2 standard, when I change something I have to release it as breaking change. This is clearly a bug, I'm agreed with you, but it's not so dangerous to add a possibile breaking change disclaimer in the changelog. I don't want to release it as 6 or 5 with a possibile breaking change warning.
Also it is possibile that some users changed margin and other stuff via css and I don't want to create so many problems for 5px of marginBottom.

I have to think to all users. 

I don't know if you are agree with me.

**Comment by Ks89** [comment]
Obviously this will be different for version 6. In that occasion I could change margins as I prefer.

PS maybe a warning in the doc website could be useful.

**Comment by Ks89** [comment]
@tobi-or-not-tobi What do you think? Can I release it as stable?

I changed APIs in beta2 removing 'margin' sub-object to flat everything.

All params are optional, but obviously, color, background e bottomMargin have default values to prevent breaking changes.

**Comment by tobi-or-not-tobi** [comment]
LGTM, I just would have expect that the default values would only be turned of when they got overridden. I f I now add an empty style attribute, the background color is gone. No issue at all, just not what i expected.

**Comment by Ks89** [comment]
Please could add an example of input and what do you expect as result? Because I don't undestand.

**Comment by Ks89** [comment]
Ohh yes, wait. I understood.

Damn, this is a bug. I'll fix it

**Comment by Ks89** [comment]
Ok, I released 5.4.0-beta.3. Now it should be ok.

When you confirm that everything is ok, I'll release it as final.

**Comment by tobi-or-not-tobi** [comment]
Yes, now it works as expected. Thanks!

**Comment by Ks89** [comment]
released as 5.4.0 final
